### PR TITLE
AT-627: Fix some Google search console errors

### PR DESCRIPTION
**Pull request by georgecrawford**
1. Sanitise text values, so that `<a ... data-vars-link-text="one device for " convenience""="">one device for "convenience"</a>` becomes `<a ... data-vars-link-text="one device for convenience">one device for "convenience"</a>`
2. Remove links with empty `href` or an invalid protocol (as per https://github.com/ampproject/amphtml/blob/master/validator/validator-main.protoascii)

### CODE REVIEW

**Comment by georgecrawford** [comment]
This is quite opinionated, but will clear up the vast majority of errors in the Google Search console.

File: server/lib/transforms/link-analytics.js [hunk 1]
```diff
@@ -1,13 +1,20 @@
 'use strict';
 
+const sanitise = text => text.replace(/[^\w ]/g, '');
```
Comment (quarterto) [inline server/lib/transforms/link-analytics.js]: How essential is it for Spoor to get accurate link text? Would replacing with entities be better?

File: server/lib/transforms/remove-invalid-links.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		throw Error('Invalid link protocol');
+	}
+};
+
+module.exports = ($, options) => {
+	$('a').each((i, el) => {
+		const $el = $(el);
+
+		try{
+			validate($el);
```
Comment (quarterto) [inline server/lib/transforms/remove-invalid-links.js]: I'd rather strip the link and report as a warning?

File: server/lib/transforms/link-analytics.js (hunk 1 above)
Comment (georgecrawford) [inline server/lib/transforms/link-analytics.js]: Not at all essential - it's for quick browsing by humans only (e.g. "which `more-ons-theme` is getting the most clicks?"). Plain text will suffice.

File: server/lib/transforms/remove-invalid-links.js (hunk 2 above)
Comment (georgecrawford) [inline server/lib/transforms/remove-invalid-links.js]: I do strip it out. I report as an error (agree that warning would be better). Or do you want a whole different approach?

Comment (quarterto) [inline server/lib/transforms/remove-invalid-links.js]: Oh haha I totally missed that part. Fine

**Pull request merged**